### PR TITLE
Add new assertions

### PR DESCRIPTION
**Pull request by stevenleggdfe**
- Adds referential integrity assertions, documentation & example code.
- Adds data freshness assertion & documentation.
- Adds table-level data freshness assertion & documentation

To test, check the SQL generated by example.js in the SL-dev branch. I haven't yet tested this on real data but the SQL generated by the JS looks valid to me.

### CODE REVIEW

**Review by alexchristensenedu — APPROVED** [review]
Happy for this to be merged. Minor comment on the Readme

File: index.js [hunk 1]
```diff
@@ -40,6 +43,7 @@ module.exports = (params) => {
     bqDatasetName,
     bqEventsTableName,
     bqEventsTableNameSpace,
+    eventsDataFreshnessDays,
```
[the remark this answers is not in the document]
Comment (stevenleggdfe) [reply index.js:46]: No, but that's a good idea I think. Let me see if I can whip something up that does that.

[the remark this answers is not in the document]
Comment (stevenleggdfe) [reply index.js:47]: @alexchristensenedu I've added these extra assertions. Let me know what you think!

**Review by alexchristensenedu — APPROVED** [review]
Nice work - looks good

**Review by JeffreyDfE — APPROVED** [review]
All good thank you, happy to approve

**Pull request merged**